restcomm/restcomm.http/src/main/java/org/restcomm/connect/http/security/SecurityFilter.java

```diff
@@ -107,4 +110,16 @@ protected void filterClosedAccounts(UserIdentityContext userIdentityContext, Str
             throw new WebApplicationException(status(Status.FORBIDDEN).entity("Provided Account is not active").build());
         }
     }
+
+    protected void validateOrganizationAccess(UserIdentityContext userIdentityContext, DaoManager daoManager, ContainerRequest cr) {
+        Organization effectiveAccountOrg = daoManager.getOrganizationsDao().getOrganization(userIdentityContext.getEffectiveAccount().getOrganizationSid());
```

Comment:
@jaimecasero The check we need according to the story is 
(Organization derived from account credentials) == (Organization derived from host part of the request)

But as an extra check, I can add that
(Account SID derived from account credentials) == (Account SID from request URI)

what do you think?

Reply:
I thought Charles ticket was about...:
 (Organization derived from host part of the request) == (Organization derived from Account SID from request URI)

I would like to understand requirement, and add the minimal authorization rules....

I remember there were two tickets, maybe they referred to these two conditions:
1. (Organization derived from account credentials) == (Organization derived from host part of the request) -> this is clearly 407
2. (Organization derived from host part of the request) == (Organization derived from Account SID from request URI) -> maybe this should be 404, i dont know...

can we clarify with him and PO?

Reply:
by the way, (Account SID derived from account credentials) == (Account SID from request URI)  -> i think this will be two restrictive i we take into account account hierarchy, and superadmin role...

Reply:
@jaimecasero agree about been too restrictive the  (Account SID derived from account credentials) == (Account SID from request URI)